includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php

```diff
@@ -561,6 +562,14 @@ public function audience_segments( $post_id, $compound_interest_id, $tag_ids ) {
 				__( 'Mailchimp campaign ID not found.', 'newspack-newsletters' )
 			);
 		}
+
+		if ( '' !== $target_id && ! $interest_id && ! $segment_id ) {
```

Comment:
If the condition on line 550 is not met, `$interest_id` will not be defined. It should either be defined before the condition, or its existence checked here.

Reply:
Nice catch! 8aba70be868cc7d36a43c1c05786838e9a95d8f5 fixes it.